### DISCUSSION TITLE
Retrieving JSON list of meetings from a TSML site

### DISCUSSION THREAD

**Discussion by jfieser** [code4recovery/12-step-meeting-list#475] [Q&A] [answered]
I am hoping to set up a JSON retrieval of meetings from an existing TSML site http://www.aacomm.org/.  The plan is to cache that JSON on the site I manage https://www.d12aa.org so that our subset of meetings can be displayed there.  Updates of the JSON would occur upon request when the list is older than ~10 min.  

Viewers of our subset of meetings on d12aa.org would see only those within our district either by Longitude/Latitude or by selecting individual meetings within our district. We may want to include some other information on a meeting to be displayed on our site depending on today's date i.e. if childcare is available this week or if it is an open birthday meeting since it's the last Saturday of the month.  

I see that someone else had a similar question project a couple years ago:  https://wordpress.org/support/topic/structure-of-exported-json-file/

My questions are: 1) what is the URL of the JSON feed in TSML? and 2) is there a good guid to specify individual meetings in the JSON feed?

**Comment by tech2serve** [comment, marked as answer]
Hi @jfieser --

If you search the TSML repo you'll find several discussions (mostly in the Issues today) about how to access/determine the URL of the JSON feed from TSML. The gist is to make your feed public (Sharing -> Open) on the Import & Settings page, and then navigate to the link that shows up under Public Feed:

<img width="274" alt="image" src="https://user-images.githubusercontent.com/31904316/127550049-ce697ad7-f306-4685-a33d-7d30d25db45b.png">

We do not have a guide or a process for filtering the meetings in the feed. This is something you'd have to write (are you a developer?)

We are having some convo about changing the entire process about how we share meetings between sites. Frankly, Wordpress doesn't readily support automation of this process and many have been unhappy with this fact.

**Reply by jfieser** [reply]
I see now that TSML includes the import JSON feature and have determined that my problem is that the feed I was looking for is actually not a ubiquitous Wordpress TSML site, but written by a local developer in C#/MVC.  I can code a little, but I don't feel called to "develop" for a single entity like that.  

Don't go throwing the Wordpress baby out with the bathwater.   So what if JSON feeds supply all the data in the tank?  Simply size the tanks appropriately by having each district in it's own "tank".  Apparently multi-site Wordpress (1 site for each district in eamo.org) does that just fine.  The aggregated data can be filtered at a cached aggregation without changing how we are ALL ENTERING OUR MEETING LISTS!

**Reply by tech2serve** [reply]
If the site you're wanting to get meetings from (assuming they're working with you on this) has the meetings in a database (most likely), then a script can readily be used to pull the meetings out in JSON. We have numerous examples that can be used as a starting point, although most likely the site is already using one if they provide meetings to Meeting Guide.

**Reply by jfieser** [reply]
They are working with me, but they also seek to replace the existing webmaster with someone who also has C#/MVC experience.  I don't think that is sustainable and I'd rather hitch my wagon to a longer term, widely supported solution. WP/TSML seems like just the tool. As the OP I thought it WAS WP/TSML, and your solution was spot on.  Now I find that it is not WP, and I'm ready to make it so.  
Thanks!
